src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-logic.cpp

```diff
@@ -0,0 +1,1048 @@
+        return std::nullopt;
+    }
+
+    if (transportConfiguration->transportStatus == TransportStatusEnum::kInactive)
+    {
+        auto clusterStatus = to_underlying(StatusCodeEnum::kInvalidTransportStatus);
+        ChipLogError(Zcl, "HandleManuallyTriggerTransport[ep=%d]: Invalid Transport status", mEndpointId);
+        handler.AddClusterSpecificFailure(commandPath, clusterStatus);
+        return std::nullopt;
+    }
+    if (transportConfiguration->transportOptions.HasValue())
+    {
+        if (transportConfiguration->transportOptions.Value().triggerOptions.triggerType == TransportTriggerTypeEnum::kContinuous)
+        {
+
+            auto clusterStatus = to_underlying(StatusCodeEnum::kInvalidTriggerType);
+            ChipLogError(Zcl, "HandleManuallyTriggerTransport[ep=%d]: Invalid Trigger type", mEndpointId);
+            handler.AddClusterSpecificFailure(commandPath, clusterStatus);
+            return std::nullopt;
+        }
+        if (transportConfiguration->transportOptions.Value().triggerOptions.triggerType == TransportTriggerTypeEnum::kCommand &&
+            !timeControl.HasValue())
+        {
+
+            ChipLogError(Zcl, "HandleManuallyTriggerTransport[ep=%d]: Time control field not present", mEndpointId);
+            handler.AddStatus(commandPath, Status::DynamicConstraintError);
+            return std::nullopt;
+        }
+    }
+
+    // When trigger type is motion in the allocated transport but triggering it manually
+    if (!timeControl.HasValue())
+    {
+        timeControl = transportConfiguration->transportOptions.Value().triggerOptions.motionTimeControl;
+    }
+
+    // Call the delegate
+    status = mDelegate->ManuallyTriggerTransport(connectionID, activationReason, timeControl);
+
+    if (status == Status::Success)
+    {
+        GeneratePushTransportBeginEvent(connectionID, TransportTriggerTypeEnum::kCommand, MakeOptional(activationReason));
+    }
+
+    handler.AddStatus(commandPath, status);
+
+    return std::nullopt;
+}
+
+std::optional<DataModel::ActionReturnStatus>
+PushAvStreamTransportServerLogic::HandleFindTransport(CommandHandler & handler, const ConcreteCommandPath & commandPath,
+                                                      const Commands::FindTransport::DecodableType & commandData)
+{
+    if (IsNullDelegateWithLogging(commandPath.mEndpointId))
+    {
+        return Status::UnsupportedCommand;
+    }
+
+    Commands::FindTransportResponse::Type response;
+
+    Optional<DataModel::Nullable<uint16_t>> connectionID = commandData.connectionID;
+
+    std::vector<Structs::TransportConfigurationStruct::Type> transportConfigurations;
+
+    if (!connectionID.HasValue() || connectionID.Value().IsNull())
+    {
+        if (mCurrentConnections.size() == 0)
+        {
+            ChipLogError(Zcl, "HandleFindTransport[ep=%d]: ConnectionID not found", mEndpointId);
+            handler.AddStatus(commandPath, Status::NotFound);
```

Comment:
So I'm a little curious: How did you decide on `return Status::UnsupportedCommand` above vs doing `AddStatus()` here?  Is there a reason for the difference?